pkg/vendored/controller_config.go

```diff
@@ -0,0 +1,372 @@
+	// cannot be resolved, the reference in the input string will be unchanged. The $(VAR_NAME) syntax
+	// can be escaped with a double $$, ie: $$(VAR_NAME). Escaped references will never be expanded,
+	// regardless of whether the variable exists or not.
+	// Cannot be updated.
+	// More info: https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#running-a-command-in-a-shell
+	// +optional
+	Args []string `json:"args,omitempty"`
+	// List of sources to populate environment variables in the container.
+	// The keys defined within a source must be a C_IDENTIFIER. All invalid keys
+	// will be reported as an event when the container is starting. When a key exists in multiple
+	// sources, the value associated with the last source will take precedence.
+	// Values defined by an Env with a duplicate key will take precedence.
+	// Cannot be updated.
+	// +optional
+	EnvFrom []corev1.EnvFromSource `json:"envFrom,omitempty"`
+	// List of environment variables to set in the container.
+	// Cannot be updated.
+	// +optional
+	Env []corev1.EnvVar `json:"env,omitempty"`
+	// List of container ports to expose on the container
+	// +optional
+	Ports []corev1.ContainerPort `json:"ports,omitempty"`
+	// List of volumes that can be mounted by containers belonging to the pod.
+	// More info: https://kubernetes.io/docs/concepts/storage/volumes
+	// +optional
+	Volumes []corev1.Volume `json:"volumes,omitempty"`
+	// List of VolumeMounts to mount into the container's filesystem.
+	// Cannot be updated.
+	// +optional
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+}
+
+// PodObjectMeta is metadata that is added to the Pods in a provider's
+// Deployment.
+type PodObjectMeta struct {
+	// Annotations is an unstructured key value map stored with a resource that may be
+	// set by external tools to store and retrieve arbitrary metadata. They are not
+	// queryable and should be preserved when modifying objects.
+	// More info: http://kubernetes.io/docs/user-guide/annotations
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Map of string keys and values that can be used to organize and
+	// categorize (scope and select) objects. This will only affect
+	// labels on the pod, not the pod selector. Labels will be merged
+	// with internal labels used by crossplane, and labels with a
+	// crossplane.io key might be overwritten.
+	// More info: http://kubernetes.io/docs/user-guide/labels
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+// ControllerConfig is the CRD type for a packaged controller configuration.
+// Deprecated: This API is replaced by DeploymentRuntimeConfig, and is scheduled
+// to be removed in a future release. See the design doc for more details:
+// https://github.com/crossplane/crossplane/blob/11bbe13ea3604928cc4e24e8d0d18f3f5f7e847c/design/one-pager-package-runtime-config.md
+type ControllerConfig struct {
```

Comment:
What is the reason for the deprecated `ControllerConfig` instead of `DeploymentRuntimeConfig`? To support older crossplane versions for now?

Reply:
Exactly. Lets follow up with #23